### ISSUE TITLE
_Table.Definition.cshtml wrong boolean value check for checkBox & renderBoolean

### ISSUE THREAD

**Issue by DevRebl** [nopSolutions/nopCommerce#4183]
nopCommerce version: 4.20

Steps to reproduce the problem: 

While upgrading from 4.10 to 4.20 I ran into a problem. In one of the plugins the checkboxes were not checked. I looked at the `_Table.Definition.cshtml` file and found the problem why this doensn't work. 

Currently the checkbox data value is checked like this: `data === 'true'`. `data` is not a string but a boolean. Since `data` is a boolean we do not need to explicitly check if `data == true`. (this is done for the `renderBoolean` case)

Current:
```
case RenderCheckBox checkBox:
    <text>
        render: function (data, type, row, meta) {
            return (data === 'true')
                ? '<input name="@checkBox.Name" value="' + row.@checkBox.PropertyKeyName + '" type="checkbox" class="checkboxGroups" checked="checked" />'
                : '<input name="@checkBox.Name" value="' + row.@checkBox.PropertyKeyName + '" type="checkbox" class="checkboxGroups" />';
        },
    </text>
break;
case RenderBoolean renderBoolean:
    <text>
        render: function (data, type, row) {
            return (data == true)
                ? '<i class="fa fa-check true-icon" nop-value="true"></i>'
                : '<i class="fa fa-close false-icon" nop-value="false"></i>';
        },
    </text>
break;
```

Suggested fix:
```
case RenderCheckBox checkBox:
    <text>
        render: function (data, type, row, meta) {
            return data
                ? '<input name="@checkBox.Name" value="' + row.@checkBox.PropertyKeyName + '" type="checkbox" class="checkboxGroups" checked="checked" />'
                : '<input name="@checkBox.Name" value="' + row.@checkBox.PropertyKeyName + '" type="checkbox" class="checkboxGroups" />';
        },
    </text>
break;
case RenderBoolean renderBoolean:
    <text>
        render: function (data, type, row) {
            return data
                ? '<i class="fa fa-check true-icon" nop-value="true"></i>'
                : '<i class="fa fa-close false-icon" nop-value="false"></i>';
        },
    </text>
break;
```

**Comment by DmitriyKulagin** [comment]
@DevRebl It seems that you are trying to use the RenderCheckBox not as it was originally conceived. It is not used to display table data.

In fact, "RenderCheckBox" is supposed to be used only to support the functionality of "IsMasterCheckBox". Therefore, "data" in this case does not play any meaning. The component should always be rendered "unchecked" when the page is refreshed.

In the case of renderBoolean there really is a "boolean" value in "data" and the check (data == true) can be replaced with "data" without breaking the logic.

**Comment by DmitriyKulagin** [comment]
Closed #4183
